director/stat_stress_test.go

```diff
@@ -54,61 +114,149 @@ var (
+	// specific to this unit test:
+	// - PELICAN_STRESS_DURATION: Parsed as a time duration, this is the time of the
+	//   stress test loop.
+	// - PELICAN_STRESS_SNAPSHOT_HEAP: If set to `1`, then a snapshot of the in-use
+	//   heap will be written to disk before and after the main stress test; the desire
+	//   is to allow developers to compare the contents with `go tool pprof`
+	testDuration := 10 * time.Second
+	testDurationEnv := os.Getenv("PELICAN_STRESS_DURATION")
+	if testDurationEnv != "" {
+		var err error
+		testDuration, err = time.ParseDuration(testDurationEnv)
+		require.NoError(t, err)
+	}
+	snapshotHeap := os.Getenv("PELICAN_STRESS_SNAPSHOT_HEAP") == "1"
+
 	viper.Set(param.Xrootd_EnableLocalMonitoring.GetName(), false)
-	viper.Set(param.Server_AdLifetime.GetName(), "500ms")
+	// Under testing on a laptop, we saw up to 1 second long delays happen deep in the
+	// go HTTP server framework (GC?  Lock contention?  Unclear...).  By bumping the ad
+	// lifetime to 2 seconds, we get repeated ad updates through the lifetime of the
+	// test - useful for checking propagation of changes - but don't get the spurious
+	// failures if there's a short blip in availability.
+	viper.Set(param.Server_AdLifetime.GetName(), "2000ms")
 	viper.Set(param.Cache_SelfTest.GetName(), false)
 	viper.Set(param.Origin_DirectorTest.GetName(), false)
 	viper.Set(param.Origin_SelfTest.GetName(), false)
+	viper.Set(param.Director_CachePresenceCapacity.GetName(), 500)
 	fed := fed_test_utils.NewFedTest(t, directorPublicCfg)
 	config.DisableLoggingCensor()
 	discoveryUrl, err := url.Parse(param.Federation_DiscoveryUrl.GetString())
 	assert.NoError(t, err)
 
+	require.NoError(t, config.InitClient())
+
 	grp, _ := errgroup.WithContext(fed.Ctx)
 	grp.SetLimit(10)
 	idx := 0
 	start := time.Now()
-	dest := filepath.Join(t.TempDir(), "dest.txt")
 	cacheSize := param.Director_CachePresenceCapacity.GetInt()
 
+	testdir := filepath.Join(fed.Exports[0].StoragePrefix, "stress")
+	require.NoError(t, os.Mkdir(testdir, os.FileMode(0700)))
+	ui, err := config.GetDaemonUserInfo()
+	require.NoError(t, err)
+	require.NoError(t, os.Chown(testdir, ui.Uid, ui.Gid))
+
 	// Fill the cache before taking the baseline measurement. Otherwise,
 	// it might end up that increased memory usage is due to filling up the
 	// cache and not an actual memory leak.
-	for idx < cacheSize {
+	//
+	// We make sure the warmup is for at least the size of the cache and 10%
+	// the length of the entire test duration.
+	isCanceled := false
+	warmupDuration := testDuration / 10
+	for idx < cacheSize || time.Since(start) < warmupDuration {
 		downloadURL := fmt.Sprintf("pelican://%s%s/stress/%v.txt", discoveryUrl.Host, fed.Exports[0].FederationPrefix, idx)
+		destName := filepath.Join(t.TempDir(), fmt.Sprintf("dest.%v.txt", idx))
+		src := filepath.Join(fed.Exports[0].StoragePrefix, fmt.Sprintf("stress/%v.txt", idx))
+		fmt.Println(src)
```

Comment:
Do you mean to have this? If so, can we change it to a `test.Log` line?

Reply:
I think we can do without - I'll delete.